### PR TITLE
e2e: setup testing for multi-version

### PR DESCRIPTION
**Pull request by williambanfield**
This pull requests sets up the e2e tests to be able to support multiple versions within the same test network. This is achieved through a few simple changes:

* Each node takes a `version` parameter in the testnet manifest. This dictates which version of the testapp to use. If not set, the locally available version is used.
* Adds a `testapp-docker.yml` workflow that publishes the testapp to docker hub so that tagged versions may be available for use in a multi-version test network.

This change does not actually add a testnetwork that does multi-version testing. Since no previous versions of the testapp have been published to dockerhub, there are not old versions available to test against. We'll either need to configure this after the next minor release which will trigger a testapp to be pushed to dockerhub, or push an image from the previous version of Tendermint so that the multiversion test has an old version to pull.

#### PR checklist

- [ ] Tests written/updated, or no tests needed
- [ ] `CHANGELOG_PENDING.md` updated, or no changelog entry needed
- [ ] Updated relevant documentation (`docs/`) and code comments, or no
      documentation updates needed

### CODE REVIEW

File: .github/workflows/testapp-docker.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            VERSION=$(echo ${GITHUB_REF#refs/heads/} | sed -r 's#/+#-#g')
+            if [ "${{ github.event.repository.default_branch }}" = "$VERSION" ]; then
+              VERSION=latest
+            fi
+          fi
+          TAGS="${DOCKER_IMAGE}:${VERSION}"
+          if [[ $VERSION =~ ^v[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
+            TAGS="$TAGS,${DOCKER_IMAGE}:${VERSION}"
+          fi
+          echo ::set-output name=tags::${TAGS}
```
Comment (thanethomson) [inline .github/workflows/testapp-docker.yml:36]:
```suggestion
          echo "tags=${TAGS}" >> $GITHUB_OUTPUT
```

As per https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

File: .github/workflows/testapp-docker.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+name: Docker
```
Comment (thanethomson) [inline .github/workflows/testapp-docker.yml:1]:
```suggestion
name: Testapp Docker
```

**Review by thanethomson — APPROVED** [review]
Just a couple of minor things, but otherwise looks good. I assume this needs to be backported to v0.37 and v0.34?

**Comment by williambanfield** [comment]
> Just a couple of minor things, but otherwise looks good. I assume this needs to be backported to v0.37 and v0.34?

Yeah, it will need a backport.

**Review by thanethomson — APPROVED** [review]

**Pull request merged**